### ISSUE TITLE
Create a new filesystem on a SD card

### ISSUE THREAD

**Issue by glebone** [CypherpunkArmory/UserLAnd#160]
Hello! 

Is it possible to create a new filesystem on a External storage? 
Or add the ability to move the application to the external SD card entirely? 
Thanks a lot!

**Comment by lapsio** [comment]
That's something I'd also like to address.

I'm aware there's significant problem with sdcard filesystems because afaik Android doesn't allow to change permissions or modification dates on sdcard (to the past, modification to current date should work) and it's something GNURoot never managed to solve either, but maybe something changed in this matter?

**Comment by glebone** [comment]
This feature worked in GNU Root Debian, and I really miss SD card image storing feature there. Since UserLAnd supports multi filesystem, it simply drains my device internal memory.

**Comment by lapsio** [comment]
@glebone GNU Root Debian always suffered serious issues with permissions so it's hard to call it "working feature". For example making script stored in /home executable on permissions level (0777) was impossible iirc. It had some basic permissions emulation for other mountpoints but basically permissions and modification dates handling was mess. A lot of programs which required modification date or permission changes simply crashed under GNURoot. That includes many installation scripts which were calling chmod 755 on script and then executing it as executable file. It's something I didn't experience with UserLAnd, even though I installed the same programs in the same way. It's much more compatible on chmod / chown / utime level with actual linux. 

I believe it's exactly because files are not stored on sdcard so probably UserLAnd has more control over permissions on Android level than it would have if files were stored on sdcard.

**Comment by glebone** [comment]
@lapsio  Thanks for the detailed answer! Agree, at least mounting some directories from the SD card would be useful

**Comment by corbinlc** [comment]
@projectextremum I had a working, but not well tested, Proot extension to solve this, but it was really expensive.  Effectively it copied file to internal storage when needed.  I like the idea of `userland-exec`.  Where is this project being discussed as I see a few things with that name and am not familiar with it?

**Comment by adamsson6** [comment]
Is it possible to download the linux system on sdcard ? I do not have enough internal memory so the app cannot run on my phone...

**Comment by ncodeyx** [comment]
I am also interested in this. Is there an update?

**Comment by Miqueloti** [comment]
I'm also interested too! I buy a 128GB SDXC only for this, but i can't move :)

**Comment by EoflaOE** [comment]
Same, can't move to SD Card. Critically needed for an 8GB device. I need an SD Card to enjoy effective Linux experience.

**Comment by altergui** [comment]
bumped into this limitation as well.
i'm not familiar with Android restrictions, but has anyone considered the option of creating a filesystem inside a file on the sdcard, and then mounting the file? or is that impossible in android without root?

**Comment by Wiiplay123** [comment]
Same problem here, I don't have much internal memory but I have a 64GB SD card that would be great for installing all these apps.

Would a disk image like altergui is talking about work?

**Comment by und3fined404** [comment]
I think you can in kali 
![Screenshot_20200207-215804](https://user-images.githubusercontent.com/59595165/74065521-65aeef80-49f5-11ea-94b0-e317385d7a01.png)

**Comment by up2smoke** [comment]
> I think you can in kali
> ![Screenshot_20200207-215804](https://user-images.githubusercontent.com/59595165/74065521-65aeef80-49f5-11ea-94b0-e317385d7a01.png)

Enter ls and u will see ur sdcard and internal will be emtpy. So its not a help or solution

**Comment by ciaoly** [comment]
Hey there, this is an issue in [repository AnLinux](https://github.com/EXALAB/AnLinux-App) for external storage supporting , anyone knows is it useful for UserLAnd?  

*Link:* [Feature Request: storage #85](https://github.com/EXALAB/AnLinux-App/issues/85#issuecomment-594995824)

**Comment by GrpeApple** [comment]
Can't UserLand just store files in <SD Card>/Android/data/files/tech.ula/files like in Termux? Or am I being ignorant that I can't change permissions the same as in Termux?

**Comment by niccokunzmann** [comment]
> creating a filesystem inside a file on the sdcard, and then mounting the file

Yes, I considered. I am looking at the issues to find out how to mount this.

**Comment by niccokunzmann** [comment]
I found that I could not use mount because my phone is not rooted - at least that is what I think why it does not work.
So, I created a tool [sdlink](https://gitlab.com/niccokunzmann/sdlink/) that links many of my files onto the sd card. It works for me, using Python and NodeJS. I am happy to hear other use-cases and see if we can make them work.
But: I did not try [creating a ext partition on my sdcard](https://github.com/EXALAB/AnLinux-App/issues/85#issuecomment-594995824) as mentioned by [ciaoly](https://github.com/ciaoly).

**Comment by corbinlc** [comment]
Closing as part of a repository cleanup. If this issue is still relevant, feel free to reopen it or file a fresh one.
